apollo-router/src/plugins/telemetry/otlp.rs

```diff
@@ -156,50 +161,51 @@ pub(super) fn process_endpoint(
 impl Config {
     pub(crate) fn exporter<T: From<HttpExporterBuilder> + From<TonicExporterBuilder>>(
         &self,
-        kind: TelemetryDataKind,
-    ) -> Result<T, BoxError> {
-        match self.protocol {
-            Protocol::Grpc => {
-                let endpoint_opt = process_endpoint(&self.endpoint, &kind, &self.protocol)?;
-                // Figure out if we need to set tls config for our exporter
-                let tls_config_opt = if let Some(endpoint) = &endpoint_opt {
-                    if !endpoint.is_empty() {
-                        let tls_url = Uri::try_from(endpoint)?;
-                        Some(self.grpc.clone().to_tls_config(&tls_url)?)
-                    } else {
-                        None
-                    }
-                } else {
-                    None
-                };
-
-                let mut exporter = opentelemetry_otlp::new_exporter()
-                    .tonic()
-                    .with_protocol(opentelemetry_otlp::Protocol::Grpc)
-                    .with_timeout(self.batch_processor.max_export_timeout)
-                    .with_metadata(MetadataMap::from_headers(self.grpc.metadata.clone()));
-                if let Some(endpoint) = endpoint_opt {
-                    exporter = exporter.with_endpoint(endpoint);
-                }
-                if let Some(tls_config) = tls_config_opt {
-                    exporter = exporter.with_tls_config(tls_config);
-                }
-                Ok(exporter.into())
-            }
-            Protocol::Http => {
-                let endpoint_opt = process_endpoint(&self.endpoint, &kind, &self.protocol)?;
-                let headers = self.http.headers.clone();
-                let mut exporter: HttpExporterBuilder = opentelemetry_otlp::new_exporter()
-                    .http()
-                    .with_protocol(opentelemetry_otlp::Protocol::Grpc)
-                    .with_timeout(self.batch_processor.max_export_timeout)
-                    .with_headers(headers);
-                if let Some(endpoint) = endpoint_opt {
-                    exporter = exporter.with_endpoint(endpoint);
-                }
-                Ok(exporter.into())
-            }
-        }
+        _kind: TelemetryDataKind,
+    ) -> Result<T, BoxError>
+    {
+        todo!("We can uncomment this when we upgrade to v0.30.0 since SpanExporterBuilder and MetricExporterBuilder will be publicly re-exported")
```

Comment:
Similar point here. Again, I think I recommended something along these lines, but we shouldn't commit panics to `dev`.